### PR TITLE
Fix issue with 'campaign_cause' transformer.

### PR DESCRIPTION
**Pull request by DFurnes**
### What's this PR do?

This pull request fixes an issue where `$campaign` not existing causes `implode` to pout when it's passed `null`. As a quick fix, we'll use the `?:` operator to pass an empty array in this scenario.

### How should this be reviewed?

👀

### Any background context you want to provide?

This is happening because we're relying on the `optional()` helper's fallback behavior, and not the "empty" array provided by this field's accessor. (I also considered using a `$campaign ? … : null` ternary, but that made this take up multiple lines.)

### Relevant tickets

References [Pivotal #175209573](https://www.pivotaltracker.com/story/show/175209573) and [this Slack thread](https://dosomething.slack.com/archives/CUQMTP0RM/p1602278321008400?thread_ts=1602278269.008300&cid=CUQMTP0RM).

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: tests/Unit/Models/PostModelTest.php [hunk 1]
```diff
@@ -153,6 +153,23 @@ public function testBlinkPayload()
...
+     */
+    public function testCustomerIoPayloadWithoutSignup()
+    {
+        $post = factory(Post::class)->create();
+        $post->signup->delete();
+
+        $payload = $post->fresh()->toCustomerIoPayload();
+
+        $this->assertArrayHasKey('id', $payload);
+        $this->assertEquals('', $payload['campaign_cause']);
```
Comment (DFurnes) [inline tests/Unit/Models/PostModelTest.php:170]: I should've done this from the get-go 🦎 but this should give us some peace of mind that this doesn't regress in the future (since it's so easy to forget this edge case). The main value of this test is to see whether `toCustomerIoPayload` throws an exception, so I haven't filled out too many assertions besides that.

File: app/Models/Post.php [hunk 2]
```diff
@@ -366,7 +366,7 @@ public function toCustomerIoPayload()
                 'campaign_id' => (string) $this->campaign_id,
                 'campaign_title' => Arr::get($campaignWebsite, 'title'),
                 'campaign_slug' => Arr::get($campaignWebsite, 'slug'),
-                'campaign_cause' => implode(',', $campaign->cause),
+                'campaign_cause' => implode(',', $campaign->cause ?: []),
```
Comment (DFurnes) [inline app/Models/Post.php:369]: For context, here's the error we were getting on this before (when`$campaign->cause` was `null`):

```
production.ERROR: implode(): Invalid arguments passed
```

**Review by katiecrane — APPROVED** [review]
oh tricky!! I really like that you added a test for this case too!
